akka-stream/src/main/scala/akka/stream/impl/ActorMaterializerImpl.scala

```diff
@@ -229,34 +225,78 @@ private[akka] class SubFusingActorMaterializerImpl(
   implicit val ec: ExecutionContextExecutor = context.dispatcher
   override def supervisorStrategy: SupervisorStrategy = SupervisorStrategy.stoppingStrategy
 
+  private var snapshotCollectors: Set[ActorRef] = Set.empty
+
   def receive: Receive = {
     case Materialize(props, name) =>
       val impl = context.actorOf(props, name)
       sender() ! impl
     case GetChildren =>
       sender() ! Children(context.children.toSet)
-    case GetChildrenSnapshots =>
-      takeSnapshotsOfChildren().map(ChildrenSnapshots.apply _).pipeTo(sender())
-
+    case GetChildrenSnapshots(timeout) =>
+      val collector =
+        context.actorOf(
+          SnapshotCollector
+            .props(context.children.toSet -- snapshotCollectors, timeout, sender())
```

Comment:
`diff` is supposed to be better than `--` for Sets

Reply:
If other actors are named we could get away without the `snapshotCollectors` collection by filtering out children with name starting with `$`. Maybe too sneaky.

Reply:
I think they all are anonymous, and I didn't want to introduce a counter to keep collector children names unique.

Reply:
Oh wait, I'm wrong, the name comes from the materialize command. Hmm. Maybe this is more clear anyway.